cmd/esmapping-generator/app/flags.go

```diff
@@ -66,6 +68,16 @@ func (o *Options) AddFlags(command *cobra.Command) {
 		"jaeger-ilm-policy",
 		"The name of the ILM policy to use if ILM is active")
 
+	o.Indices.Spans.TemplateNumShards = indexOptions.TemplateNumShards
```

Comment:
AddFlags only performs registration. What effect do you expect from these assignments? It looks like copying of zero values.

Do this config and this function even need to change?